Update release process docs (#747)
flowforge/website: src/handbook/development/releases/process.md

---
# Release Process

We do two types or releases:

>  - Scheduled releases at a regular [monthly cadence](./planning.md#cadence). This
>    encompasses the entire set of product repositories.
>  - Fix releases of individual repositories as needed

Reviewer: ```suggestion
 - [Scheduled Releases](#scheduled-releases) - _for the planned release of the entire set of product repositories._
 - [Fix Releases](#fix-releases) - _for ad-hoc release of individual repositories._
```

(Suggestion)

1. Link to the sections mentioned 
2. Remove the link to planned cadence as it is linked immediately below in the afore mentioned section

>  - Scheduled releases at a regular [monthly cadence](./planning.md#cadence). This
>    encompasses the entire set of product repositories.
>  - Fix releases of individual repositories as needed

Reviewer: ```suggestion
 - [Scheduled Releases](#scheduled-releases) - _for the planned release of the entire set of product repositories._
 - [Fix Releases](#fix-releases) - _for ad-hoc release of individual repositories._
```

(Suggestion)

1. Link to the sections mentioned 
2. Remove the link to planned cadence as it is linked immediately below in the actual section

## Scheduled Releases

> We publish a new release of FlowForge [every four weeks](./planning.md#cadence). Each
> release has a Release Manager who oversees the process. This role rotates around
> the team to ensure everyone is able to do it and that we don't have any single
> points of failure in the process.

Reviewer: ```suggestion
We publish a new release of FlowForge in line with our [release plan](./planning.md#cadence). Each
release has a Release Manager who oversees the process. This role rotates around
the team to ensure everyone is able to do it and that we don't have any single
points of failure in the process.
```

(suggestion)

Removing the words "every four weeks" means any change to cadence section negates the need for a secondary update that could potentially be missed.

### Setup

The Release Manager should take the following steps at the *start* of the release
iteration:

  - Create a Release checklist issue using the [Release Checklist template](https://github.com/flowforge/admin/issues/new?assignees=&labels=&template=release.md&title=Release%3A){rel="nofollow"}.
 - Assign the issue to the Release Manager
 - Ensure the following tools are installed and working:
    - git command-line tools
      - Ensure ssh key access is setup so git interactions do not require continual
        password typing - `ssh -T git@github.com`
    - [GitHub client](https://github.com/cli/cli)
      - Ensure the client is logged in using `gh auth login`
    - [jq](https://stedolan.github.io/jq/download/)
    - [yq](https://mikefarah.gitbook.io/yq/#install)

> On release day the Release Manager takes the lead on running the release process,
> following the steps in the checklist.

Reviewer: ```suggestion
On release day, the Release Manager takes the lead in running the release process, following the steps in the checklist.
```

I propose a comma after "On release day" to separate the introductory clause from the main clause

The release is split into four phases:

1. Phase One: Check parts not covered by the automation are up to date.
2. Phase Two: Run the release automation and verifiy each stage completes properly

> 3. Phase Three: Steps to deploy the release to Staging/Production environments
> 4. Phase Four: Publicise the release

Reviewer: ```suggestion
3. Phase Three: Steps to deploy the release to Staging/Production environments.
4. Phase Four: Publicise the release.
```

Full-stop consistency with Phase One

 - [`flowforge/flowforge-nr-project-nodes`](https://github.com/flowforge/flowforge-nr-project-nodes)
 - [`flowforge/flowforge-nr-file-nodes`](https://www.github.com//flowforge/flowforge-nr-file-nodes)
 - [`flowforge/flowforge-nr-persistent-context`](https://github.com/flowforge/flowforge-nr-persistent-context)
 - [`flowforge/flowforge-nr-tools-plugin`](https://www.github.com//flowforge/flowforge-nr-tools-plugin)

Verify [`flowforge/flowforge-nr-launcher`](https://www.github.com//flowforge/flowforge-nr-launcher) `package.json` has been updated to pull in the latest versions of `file-nodes`/`persistent-context`/`project-nodes`.

If any need to be updated, refer to the section [Unmanaged Releases](#unmanaged-releases) for releasing these.

### Phase Two

There are three scripts used run the release

  - `checkout-release` - ensures a clean copy of all required code is checked out locally
  - `prepare-release` - publishes a PR against each repository that updates the version number and changelogs as needed
  - `create-release` - creates the GitHub Releases

Follow these steps to run the scripts:

> 1. Checkout the `flowforge/admin` repository if you do not already have it.
>    Ensure you have the latest with a `git pull`.

Reviewer: ```suggestion
1. Clone the [`flowforge/admin`](https://github.com/flowforge/admin) repository if you do not already have it.
   Ensure you have the latest with a `git pull`.
```

Slightly more accurate?

4. Run the next script in the process:
   ```bash
   ../admin/create-release 1.x.y
   ```
   For each repository in the release process it:
    - Checks all npm dependencies have been published. If any are missing it blocks
      until the user asks it to check again.
    - Checks if there are any open PRs in the repository. This will include the
      Release PR created in the previous step.
      The Release Manager should get the Release PR (and any others that have been
      missed at this late state) reviewed and merged before allowing the script to continue.
      This includes ensuring the tests are passing cleanly.
      **Note:** This will require a second person to do the review as the RM cannot
      approve their own PRs.
    - The script will then create a *draft* GitHub release and provide its URL.
      The Release Manager should review the draft - check the changelog looks
      appropriate and take note of any first-time contributors to provide to the
      release blog post author.
    - The script will then publish the release and move onto the next repository.

### Phase Three

Once everything has been published, the Release Manager should:

> 1. Run [staging CI pipeline publish](https://github.com/flowforge/CloudProject/actions/workflows/build-kube.yml), to ensure staging is running the latest release.

Reviewer: ```suggestion
1. Run the staging CI pipeline [Build for Kube](https://github.com/flowforge/CloudProject/actions/workflows/build-kube.yml) against the main branch, to ensure staging is running the latest release.
```

Clarify the name for anyone suprised to see "Build for Kube" instead of "staging CI pipeline" when opening the link.

> 1. Run [staging CI pipeline publish](https://github.com/flowforge/CloudProject/actions/workflows/build-kube.yml), to ensure staging is running the latest release.

Reviewer: ```suggestion
1. Run the staging CI pipeline [Build for Kube](https://github.com/flowforge/CloudProject/actions/workflows/build-kube.yml) against the main branch, to ensure staging is running the latest release.
```

Clarify the name for anyone suprised to see "Build for Kube" instead of "staging CI pipeline" when opening the link.

Reviewer: _ignore duplicate suggestion (GH was acting up while doing review)_
